components/infra-proxy-service/migrations/pipeline/utility.go

```diff
@@ -738,3 +715,70 @@ func createLocalUser(ctx context.Context, localUserClient local_user.UsersMgmtSe
 	}
 	return nil
 }
+
+// StoreUsers reads the Result struct and populate the users table
+func StoreUsers(ctx context.Context, st storage.Storage, localUserClient local_user.UsersMgmtServiceClient, res pipeline.Result) (pipeline.Result, error) {
+	var err error
+	var totalSucceeded, totalSkipped, totalFailed int64
+
+	for _, user := range res.ParsedResult.Users {
+		_, err = StoreUser(ctx, st, user, res.Meta.ServerID, localUserClient)
+		if err != nil {
+			totalFailed++
+			continue
+		}
+		if user.ActionOps == pipeline.Skip {
+			totalSkipped++
+			continue
+		}
+		totalSucceeded++
+	}
+	if len(res.ParsedResult.Users) == int(totalFailed) {
```

Comment:
This check seems unnecessary. If the intention is to log error message I would say adding a log which logs the number of total, successful and completed users count should be good.